File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (n: int): string =
    let res = 
        [3, "Pling"; 5, "Plang"; 7, "Plong"] 
        |> List.map (fun (i, v) -> if n % i = 0 then v else "") 
        |> String.concat ""

    if res = "" then string n else res

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

drops :: [(Int, String)]
drops =
  [ (3, "Pling")
  , (5, "Plang")
  , (7, "Plong")
  ]

convert :: Int -> String
convert number =
  if null sounds 
    then show number
    else concat sounds
  where 
    factors = filter (\(x, _) -> mod number x == 0) drops
    sounds = map snd factors
